src/Type/IntersectionType.php

```diff
@@ -1029,7 +1029,10 @@ public function toArray(): Type
 	public function toArrayKey(): Type
 	{
 		if ($this->isNumericString()->yes()) {
-			return new IntegerType();
+			return new UnionType([
```

Comment:
This should be `TypeCombinator::union()` instead of `new UnionType` to ensure type normalization.